Yda 5077 eus to python rebased

stsnel opened the pull request:
Part of a pair of PRs, together with https://github.com/UtrechtUniversity/yoda-external-user-service/pull/4

stsnel committed:
YDA-5077: migrate EUS from PHP to Python 3.8

stsnel committed:
EUS: ensure API runs only on API interface

stsnel committed:
YDA-5077: ensure Python EUS works on separate host

Ensure that Python-based EUS can also work when deployed
on a separate server (as opposed to on the same server as
the web portal).

lwesterhof commented on roles/yoda_external_user_service/tasks/main.yml:
```diff
@@ -1,29 +1,66 @@
 # copyright Utrecht University
 
-- name: Install Yoda External User Service dependencies
+- name: Remove no longer used Yoda External User Service dependencies
```
Can we remove all the left over PHP packages here? php-7.4* or something similar? Or maybe use the php role to erase all traces of PHP, so it can be used by the portal and the EUS?

stsnel replied:
Have added a new remove_php role for this task, that is now a dependency for portal and EUS hosts.

lwesterhof commented on roles/yoda_external_user_service/tasks/main.yml:
```diff
@@ -32,23 +69,42 @@
-  with_items:
-    - public
-    - api
+- name: Check out Yoda portal for themes used by External User Service
```
Can we do a sparse checkout of the static and themes directory, or do we need more?

stsnel replied:
I have tested to what extent a sparse checkout improves performance or required diskspace in this situation. The sparse checkout was not faster than a regular checkout, and required diskspace was only slightly less (roughly 10% of repository size / 3 MB). I think it's best to do a regular checkout for the time being, in this case, since it enables us to use ansible.builtin.git and does not result in a severe performance/diskspace impact.

lwesterhof commented on roles/yoda_external_user_service/tasks/main.yml:
```diff
@@ -77,79 +133,70 @@
     database: "{{ eus_db_name }}"
     state: present
-    privs: SELECT,INSERT,UPDATE,DELETE
+    privs: ALL
```
Do we need all privs?

stsnel replied:
Right, these could/should be stricter. Have changed them now.

stsnel committed:
EUS: restrict database privs to minimum

stsnel committed:
Remove PHP from portal and EUS hosts

PHP was previously used in the Yoda portal and EUS,
but is no longer needed after Yoda 1.8.

stsnel committed:
EUS test users: force schema creation

Trigger creation of EUS database tables if needed, so that we can
insert values into them. These tables are created on the first API
or UI request by the EUS.

lwesterhof committed:
Merge branch 'development' into yda-5077-eus-to-python-rebased

stsnel committed:
EUS: suppress changes on various tasks

- The `git rev-head` tasks do not perform any changes, they are only
  used for extracting facts. Therefore never mark them as changed.
- Change status on pip tasks does not reflect whether any dependencies
  were upgraded. Always set to unchanged, so that users don't see
  false positive changed tasks.

stsnel committed:
Merge remote-tracking branch 'origin/yda-5077-eus-to-python-rebased' into yda-5077-eus-to-python-rebased

lwesterhof reviewed (approved):
LGTM
